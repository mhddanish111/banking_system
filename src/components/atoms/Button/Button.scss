@import '../../../styles/colors.scss';
@import '../../../styles/boarder.scss';
@import '../../../styles/spacing.scss';
@import '../../../styles/mixins.scss';
@import '../../../styles/size.scss';
@import '../../../styles/fonts.scss';
button {
    @include font-body;
    border: none;
    background-color: $color-link;
    border-radius: $border-radius;
    color: $color-font-light;
    height: $height-medium;
    padding: 0 $space-l;
    text-transform: uppercase;
    line-height: $line-height-button;
    &.active {
      outline: none;
    }
    &:hover {
      background-color: $color-link-hover;
      transition: $hover-transition;
      box-shadow: none;
      cursor: pointer;
    }
    &:disabled {
      background-color: $color-disabled;
      cursor: default;
    }
    &.dark {
      background: transparent;
      border: 1px solid $color-white;
      &:hover {
        background-color: $color-white;
        color: $color-font-medium;
        transition: $hover-transition;
      }
    }
  }