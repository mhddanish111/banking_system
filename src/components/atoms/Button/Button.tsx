import React from 'react';
import './Button.scss';

type buttonProps = {
  className?: string;
  children?: React.ReactNode | String;
  type?: 'submit' | 'reset' | 'button';
  ariaLabel?: string;
  disabled?: boolean;
  others?: any;
  onClick?: any;
};

const Button: React.FC<buttonProps> = ({
  className,
  children,
  type,
  disabled,
  ariaLabel,
  ...others
}) => (
  <button
    aria-label={ariaLabel}
    className={className}
    disabled={disabled}
    type={type}
    {...others}
  >
    {children}
  </button>
);

export default Button;
