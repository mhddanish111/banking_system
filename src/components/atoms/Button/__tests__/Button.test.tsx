import { shallow } from 'enzyme';
import Button from '../Button';

describe('<Button />', () => {
  const title = 'submit';
  const props = {
    className: '',
    children: <div />,
    ariaLabel: 'button',
    disabled: false,
    onClick: jest.fn(),
  };
  const wrapped = shallow(
    <Button {...props} type="submit">
      {title}
    </Button>,
  );
  it('should render the Title Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
