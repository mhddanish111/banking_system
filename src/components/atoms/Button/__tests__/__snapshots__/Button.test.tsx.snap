// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the Title Component correctly 1`] = `
<button
  aria-label="button"
  className=""
  disabled={false}
  onClick={[MockFunction]}
  type="submit"
>
  submit
</button>
`;
